src/emulate.c

```diff
@@ -910,6 +918,32 @@ static block_t *block_find_or_translate(riscv_t *rv)
         if (untaken == replaced_blk_entry) {
             entry->ir_tail->branch_untaken = NULL;
         }
+
+        /* upadte JALR LUT */
+        if (!entry->ir_tail->branch_table)
+            continue;
+
+#if 0
+        /*
+         * This branch lookup updating is unused since we get the PC from it and
+         * use function "cache_get()" achieve the branch prediction of T1C.
+         * However, if the structure "branch_table_t" is going to reference the
+         * block directly, this updating is nacessary to avoid to use the freed
+         * blocks.
+         */
```

Comment:
I didn't get it. When is it safe to release the resources?

Reply:
As the comments above, the current workflow is safe. However, if the implementation of `branch_table` is changed (e.g. storing the possible blocks directly instead of the program counters), this updating cannot be ignored.